src/expression.js

```diff
@@ -648,8 +648,9 @@ pp.parseIdent = function(liberal) {
   let node = this.startNode()
   if (liberal && this.options.allowReserved == "never") liberal = false
   if (this.type === tt.name) {
-    if (!liberal &&
-        (this.strict ? this.reservedWordsStrict : this.reservedWords).test(this.value) &&
+    let reservedWords = this.options.sourceType === "module" ? this.reservedWordsES6Module :
```

Comment:
Better to move comparison of `sourceType` to parser initialization so that we wouldn't be performing costly string comparison on each `Identifier` hit.